Added test about status.php

SergioBertolinSG opened the pull request:
This is a test for https://github.com/owncloud/core/issues/29228

it currently fails, requires https://github.com/owncloud/core/pull/29261 .

SergioBertolinSG committed:
Added test about status.php

phil-davis commented on tests/integration/features/status.feature:
```diff
@@ -0,0 +1,8 @@
+      When requesting status.php
+      Then the json responded should match with
+      """
+      {"installed":true,"maintenance":false,"needsDbUpgrade":false,"version":"10.0.3.3","versionstring":"10.0.3","edition":"Community","productname":"ownCloud"}
```
This will fail each time the ownCloud version number is bumped.
I did a similar sort of test for a version number in returned data in https://github.com/owncloud/core/pull/29230/files - look for "attribute value should be a valid version string" step. If you extract ``version`` and ``versionstring`` from the JSON then you can use ``version_compare()`` to discover if the data is a "reasonable valid" version number representation.

SergioBertolinSG replied:
I was thinking about checking the actual values by using occ.

phil-davis replied:
Yes, you could do that also - get those current version strings by some other way and make sure they are the same.

PVince81 reviewed (approved):
:+1: great

SergioBertolinSG committed:
Refactor of commandline and making version adaptable to time

SergioBertolinSG commented:
Ok some changes, now the version and versionstrings won't require to be changed with any newer version.

To achieve that I've refactored CommandLine trait to include all the code from CommandLineContext, removing it.

There is a variable required called ocPath, and I've added it to the constructor. This will affect all the apps, and will require to add that variable in customgroups and guests at least.

SergioBertolinSG commented:
Please rereview @PVince81 @phil-davis

SergioBertolinSG committed:
Added oc_path variable to other contexts

PVince81 commented on tests/integration/features/status.feature:
```diff
@@ -0,0 +1,8 @@
+      When requesting status.php
+      Then the status.php with versions fixed responded should match with
+      """
+      {"installed":true,"maintenance":false,"needsDbUpgrade":false,"version":"X.X.X.X","versionstring":"X.X.X","edition":"Community","productname":"ownCloud"}
```
having X in the expected version here doesn't feel right.

another approach could be to check keys individually:
"the response contains a key "installed" with value true"

in this case the version string would be checked with only:
"the response contains a key "versionString".

Also note that we have config switches to disable exposing the version, so we should also have scenarios for these

SergioBertolinSG replied:
Those X are changed with the current version of the server before comparing.

SergioBertolinSG replied:
That is what I mean with "versions fixed " in the step name.

PVince81 replied:
I know, but to someone non-technical reading the scenarios this is not self-explanatory.

If you do not want to change this to the above proposal, then I suggest changing the strings to "$CURRENT_VERSION_STRING" which most will understand that a substitution will happen there.

SergioBertolinSG replied:
OK, that sounds good.

SergioBertolinSG committed:
Changed X format for fake variables to show that there is a substitution

PVince81 commented:
@SergioBertolinSG please backport

phil-davis commented:
Backport to stable10 is in PR #29296
